src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -483,11 +489,39 @@ protected Future<Map<String, List<Piece>>> updateInventoryItemsAndHoldings(Map<S
 
     return getPoLineAndTitleById(poLineIds, requestContext)
       .compose(poLineAndTitleById -> processHoldingsUpdate(piecesGroupedByPoLine, poLineAndTitleById, requestContext)
-        .compose(v -> getItemRecords(piecesGroupedByPoLine, piecesByItemId, requestContext))
+        .compose(voidResult -> recreateItemRecords(piecesGroupedByPoLine, holder, requestContext))
+        .compose(voidResult -> getItemRecords(piecesGroupedByPoLine, piecesByItemId, requestContext))
         .compose(items -> processItemsUpdate(piecesGroupedByPoLine, piecesByItemId, items, poLineAndTitleById, requestContext))
       );
   }
 
+  protected Future<Void> recreateItemRecords(Map<String, List<Piece>> piecesGroupedByPoLine,
+                                             PiecesHolder holder,
+                                             RequestContext requestContext) {
+    if (Objects.isNull(holder.getItemsToRecreate()) || holder.getItemsToRecreate().isEmpty()) {
+      return Future.succeededFuture();
+    }
+    var futures = new ArrayList<Future<String>>();
+    piecesGroupedByPoLine.keySet().stream()
+      .map(compPolId -> holder.getItemsToRecreate().get(compPolId))
```

Comment:
Better to rename `comPolId` to `poLineId` as in other methods of the class I think, and lowercase L and 1 together could be confusing to read a bit sometimes